### PR TITLE
Added Pre- and Post-Install Hook

### LINKED ISSUE #18: [FR] Run `flux check --pre` as a pre-install hook
To ensure the cluster is compatible with Flux, we could use the `ghcr.io/fluxcd/flux-cli:v0.23.0` image in a Kubernetes Job that runs `flux check --pre`. The job could be configured as a pre-install hook.

### PR DESCRIPTION
**Pull request by MarcNor**
#### What this PR does / why we need it:
This PR will add a Pre- and Post-Install Hook for Flux to run job _flux check --pre_

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #18 

#### Checklist

- [x] [DCO](https://github.com/fluxcd-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] helm-docs are updated
- [x] Helm chart is tested

### CODE REVIEW

File: charts/flux2/templates/post-install-job.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      labels:
+        app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+        app.kubernetes.io/instance: {{ .Release.Name | quote }}
+        helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: post-install-job
+        image: "ghcr.io/fluxcd/flux-cli:v0.24.0"
+        command: ["/usr/local/bin/flux","check"]
```
Comment (stefanprodan) [inline charts/flux2/templates/post-install-job.yaml]: You need to set the namespace with `flux check --namespace <Release Namspace>`.

File: charts/flux2/templates/post-install-job.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      name: "{{ .Release.Name }}"
+      labels:
+        app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+        app.kubernetes.io/instance: {{ .Release.Name | quote }}
+        helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: post-install-job
+        image: "ghcr.io/fluxcd/flux-cli:v0.24.0"
```
Comment (stefanprodan) [inline charts/flux2/templates/post-install-job.yaml:27]: This should be exposed in values YAML like all the other images.

File: .gitignore [hunk 3]
```diff
@@ -1 +1,2 @@
 .work
+.DS_Store
```
Comment (stefanprodan) [inline .gitignore:2]: Please remove this from here and add it to your own gitignore see here how to do that https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer. Please delete all .DS_Store files from this PR.

File: charts/flux2/templates/post-install-job.yaml (hunk 1 above)
Comment (stefanprodan) [inline charts/flux2/templates/post-install-job.yaml:28]: You need to pass `--namespace <Release Namespace>`.

**Review by stefanprodan — CHANGES_REQUESTED** [review]

File: charts/flux2/templates/post-install-job.yaml (hunk 1 above)
Comment (MarcNor) [reply charts/flux2/templates/post-install-job.yaml]: Thanks for the comments! It should be resolved now!

File: charts/flux2/templates/post-install-job.yaml (hunk 2 above)
Comment (MarcNor) [reply charts/flux2/templates/post-install-job.yaml]: Thanks for the comments! It should be resolved now!

**Comment by stefanprodan** [comment]
I see you’ve checked `Helm chart is tested` but have you actually tested this?

**Comment by MarcNor** [comment]
Run with Minikube in Version 1.18.1 so that the test did fail. Afterwards did check on Version 1.21.1 so that the test did pass. Reran with the namespace to check and everything passed.

```
(base) user@pc helm-charts % kubectl logs pods/flux2-1637936378-pre-7n72j
► checking prerequisites
✔ Kubernetes 1.21.1 >=1.19.0-0
✔ prerequisites checks passed
(base) user@pc helm-charts % kubectl logs pods/flux2-1637936378-post-b26j5
► checking prerequisites
✔ Kubernetes 1.21.1 >=1.19.0-0
► checking controllers
✔ all checks passed
```
```
(base) user@pc helm-charts % kubectl logs pods/flux2-1637936953-pre-vzflf
► checking prerequisites
✗ Kubernetes version v1.18.1 does not match >=1.19.0-0
```

**Comment by haarchri** [comment]
@stefanprodan  for me LGTM - can we merge ?

**Comment by stefanprodan** [comment]
Have you disabled RBAC on Minikube? The check command does a bunch of API calls that require a service account and a cluster role binding.

**Comment by MarcNor** [comment]
Which Cluster Role Binding do you mean, that is missing? I think, there is a unit test for CLI, shouldn't that one test it?

**Comment by stefanprodan** [comment]
@MarcNor I see no service account in the job spec, but maybe Helm runs them under cluster-admin?

**Comment by stefanprodan** [comment]
Please rebase with main, to run the e2e tests.

**Comment by pierluigilenoci** [comment]
@MarcNor so what?

**Comment by stefanprodan** [comment]
@pierluigilenoci see #24

**Pull request closed without merging**